cql3/expr/prepare_expr.cc

```diff
@@ -246,6 +246,12 @@ map_prepare_expression(const collection_constructor& c, data_dictionary::databas
 
     auto key_spec = maps::key_spec_of(*receiver);
     auto value_spec = maps::value_spec_of(*receiver);
+    const map_type_impl* map_type = dynamic_cast<const map_type_impl*>(&receiver->type->without_reversed());
+    if (map_type == nullptr) {
+        on_internal_error(expr_logger,
+                          format("map_prepare_expression bad non-map receiver type: {}", receiver->type->name()));
+    }
+    data_type map_element_tuple_type = tuple_type_impl::get_instance({map_type->get_keys_type(), map_type->get_values_type()});
```

Comment:
Please adjust the commitlog to explain the bug and the fix.

Reply:
Added, forgot to write it.